src/generated/resources/openapi.json

```diff
@@ -1486,6 +1567,32 @@
         "enum" : [ "NO_TOKEN", "VALID_TOKEN", "INVALID_TOKEN", "FAILED" ],
         "type" : "string"
       },
+      "StoreType" : {
+        "enum" : [ "JKS", "PKCS12", "PEM", "UNKNOWN" ],
+        "type" : "string"
+      },
+      "TLSConfig" : {
+        "description" : "SSL configuration",
+        "required" : [ "truststore" ],
+        "type" : "object",
+        "properties" : {
+          "truststore" : {
+            "description" : "The trust store configuration for authenticating the server's certificate.",
+            "type" : "object",
+            "allOf" : [ {
+              "$ref" : "#/components/schemas/TrustStore"
+            } ]
+          },
+          "keystore" : {
+            "description" : "The key store configuration that will identify and authenticate the client to the server, required for mutual TLS (mTLS)",
+            "type" : "object",
+            "allOf" : [ {
+              "$ref" : "#/components/schemas/KeyStore"
+            } ],
+            "nullable" : true
+          }
```

Comment:
Hmm, I think this will require some discussion. If we made mTLS a credentials type for Kafka cluster, how do we support the configuration for where a user wants to use Basic/API key-secret/OAuth credentials to authenticate with Kafka over SSL?

> I feel we should, in general, differentiate between mTLS used for authentication and custom certs used for verifying the server's cert. WDYT?

Makes sense, I think. The (simplifying) assumption made was that these will most likely be the same.